lib/internal/benchmark/histogram.js

```diff
@@ -0,0 +1,207 @@
+'use strict';
+
+const { validateNumber } = require('internal/validators');
+const { codes: { ERR_OUT_OF_RANGE } } = require('internal/errors');
+const { debugBench } = require('internal/benchmark/clock');
+const {
+  MathMin,
+  MathMax,
+  MathCeil,
+  MathSqrt,
+  MathPow,
+  MathFloor,
+  NumberMAX_SAFE_INTEGER,
+  ArrayPrototypeSort,
+  ArrayPrototypePush,
+  ArrayPrototypeFilter,
+  ArrayPrototypeReduce,
+  ArrayPrototypeSlice,
+  NumberIsNaN,
+  Symbol,
+} = primordials;
+
+const kStatisticalHistogramRecord = Symbol('kStatisticalHistogramRecord');
+const kStatisticalHistogramFinish = Symbol('kStatisticalHistogramFinish');
+
+class StatisticalHistogram {
+  /**
+   * @type {number[]}
+   * @default []
+   */
+  #all = [];
+  /**
+   * @type {number}
+   */
+  #min = undefined;
```

Comment:
`= undefined;` isn't needed afaik.

```suggestion
  #min;
```